src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -41,6 +42,8 @@ public static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        private static readonly uint[] s_crcTable = Crc32ReflectedTable.Generate(0x82F63B78u);
```

Comment:
You might want to change this initializer to call a method that bails out and returns `Array.Empty<uint>()` if the hardware instruction is supported, so you don't take the cost of building the LUT when it won't be used.

Reply:
```suggestion
        private static readonly uint[] s_crcTable = Sse42.IsSupported || Crc32.IsSupported ?
            Array.Empty<uint>() : Crc32ReflectedTable.Generate(0x82F63B78u);
```
will be handled by R2R the same way as other hardware intrinsic code with software fallback.

Reply:
@kasperk81 This results in test failures ... (see CI) :/

Reply:
That's because although `Sse42.IsSupported` is true in the x86 CI environment, `Sse42.X64.IsSupported` is false.  The `ulong` overload fails that check and goes for the fallback code, only to find the LUT empty (this could have easily resulted in AV rather than just bad CRC values).

What you want to do in those cases is decompose the `ulong` down to two `uint` values and call the 32-bit intrinsic instead of falling back to the LUT version of the code.